### ISSUE TITLE
[NEW] Attach image/web preview to push notification

### ISSUE THREAD

**Issue by soundstorm** [RocketChat/Rocket.Chat.iOS#1175]
"Uploaded a file"
Or is this again a thing to open on main repository, as the notifications from the RC server just contain the text and nothing more to parse?

**Comment by rafaelks** [comment]
@soundstorm So, you're saying like previewing the first attachment of the message in the push?

**Comment by soundstorm** [comment]
@rafaelks right :) Like Twitter does

**Comment by rafaelks** [comment]
@soundstorm Sounds good to me! Could we have the first attachment of the message in the push payload @sampaiodiego?

**Comment by sampaiodiego** [comment]
by _attachment_ do you guys mean the raw `attachment` property of the message object stored on db? or do you want a special push notification data to allow image previewing?

well, in both cases the answer is yes. but for the latter we need to know which payload to send.

**Comment by rafaelks** [comment]
@sampaiodiego We'll need a property `attachment` in push for now I think and maybe another category (`MESSAGE_WITH_ATTACHMENT`) or something like that.

**Comment by sampaiodiego** [comment]
the value of the `attachment` property could be the uploaded file URL?

**Comment by rafaelks** [comment]
@sampaiodiego Yes, but now we may have a problem: how do I know what token/rid should I use to load the image? Do I know the server URL on the push payload also?

**Comment by sampaiodiego** [comment]
yes.. the server URL is already included on push payload
